x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/data_source_diff_algorithm.ts

```diff
@@ -103,7 +107,9 @@ const mergeVersions = ({
 
     case ThreeWayDiffOutcome.CustomizedValueCanUpdate: {
       if (
+        dedupedCurrentVersion &&
         dedupedCurrentVersion.type === DataSourceType.index_patterns &&
+        dedupedTargetVersion &&
         dedupedTargetVersion.type === DataSourceType.index_patterns
       ) {
```

Comment:
I think we missed this fix in the past PR:
```ts
    case ThreeWayDiffOutcome.CustomizedValueCanUpdate: {
      if (
        isIndexPatternDataSourceType(dedupedCurrentVersion) &&
        isIndexPatternDataSourceType(dedupedTargetVersion)
      ) {
        const baseVersionToMerge = isIndexPatternDataSourceType(dedupedBaseVersion)
          ? dedupedBaseVersion.index_patterns
          : [];
```
`isIndexPatternDataSourceType` needs to be exported from `three_way_diff_outcome.ts`